community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/storeview/DynamicIndexStoreView.java

```diff
@@ -71,12 +77,24 @@ public DynamicIndexStoreView( LabelScanStore labelScanStore, LockService locks,
 
     private boolean useAllNodeStoreScan( int[] labelIds )
     {
-        return ArrayUtils.isEmpty( labelIds ) || isEmptyLabelScanStore() || isNumberOfLabeledNodesExceedThreshold( labelIds );
+        try
+        {
+            return ArrayUtils.isEmpty( labelIds ) || isEmptyLabelScanStore() ||
+                    isNumberOfLabeledNodesExceedThreshold( labelIds );
+        }
+        catch ( Exception e )
+        {
+            log.error( "Can not determine number of labeled nodes, falling back to all nodes scan.", e );
```

Comment:
I was thinking that exception is already enough to figured out what went wrong and error itself is more explanation what we are doing now. Do you still think it's better to mention why it's like that and mention scan reader? @burqen